loadtest.py

```diff
@@ -5,91 +5,89 @@
-    size = (150 * 1024)
-    raw_crash, dumps = utils.generate_sized_crashes(size)
-    # Generate the payload and headers for a crash
-    payload, headers = utils.multipart_encode(raw_crash)
-    # POST the dump to Antenna
-    resp = utils.post_crash(URL_SERVER, payload, headers, compressed=True)
-    print("test_crash_150k_compressed: HTTP %s" % resp.status_code)
-    # Verify HTTP 200
-    resp.raise_for_status()
-    # Verify the response text contains a CrashID
-    utils.verify_crashid(resp.text)
+def memoize(fun):
+    """Decorates a function and memoizes it"""
+    def _memoize(*args, **kwargs):
+        key = repr((list(sorted(args)), list(sorted(kwargs.items()))))
+        if key not in CACHE:
+            ret = fun(*args, **kwargs)
+            CACHE[key] = ret
+        return CACHE[key]
+    return _memoize
 
 
-@scenario(0)
-def test_crash_400k_uncompressed():
-    size = (400 * 1024)
-    raw_crash, dumps = utils.generate_sized_crashes(size)
-    crash_payload = utils.assemble_crash_payload(raw_crash, dumps)
-    payload, headers = utils.multipart_encode(crash_payload)
-    # POST the dump to Antenna
-    resp = utils.post_crash(URL_SERVER, payload, headers)
-    print("test_crash_400k_uncompressed: HTTP %s" % resp.status_code)
-    # Verify HTTP 200
-    resp.raise_for_status()
-    # Verify the response text contains a CrashID
-    utils.verify_crashid(resp.text)
+@memoize
+def get_payload_and_headers(size, compressed=False):
+    """Returns a payload and headers
 
+    :arg size: the size in bytes the payload should be
+    :arg compressed: whether or not to compress the payload
 
-@scenario(0)
-def test_crash_4mb_uncompressed():
-    size = (4 * 1024 * 1024)
-    raw_crash, dumps = utils.generate_sized_crashes(size)
+    :returns: ``(payload, headers)``
+
+    """
+    # Generate the payload and headers for a crash
+    raw_crash, dumps = utils.generate_sized_crashes(size, compressed)
+
+    # Convert the raw crash metadata and dumps into a single dict
     crash_payload = utils.assemble_crash_payload(raw_crash, dumps)
+
+    # Convert the crash payload dict into a multipart/form-encode byte stream
     payload, headers = utils.multipart_encode(crash_payload)
 
-    if len(payload) != size:
-        raise ValueError('payload size %s', len(payload))
+    # If we should compress it, compress the byte stream and add relevant
+    # header
+    if compressed:
+        payload = utils.compress(payload)
+        headers['Content-Encoding'] = 'gzip'
+
+    # Return payload and headers as a tuple
+    return payload, headers
+
 
-    # POST the dump to Antenna
-    resp = utils.post_crash(URL_SERVER, payload, headers)
-    print("test_crash_4mb_uncompressed: HTTP %s" % resp.status_code)
+def run_test(name, size, compressed):
+    payload, headers = get_payload_and_headers(size, compressed=compressed)
+    resp = utils.post_crash(URL_SERVER, payload, headers, size)
+
+    print('%s: HTTP %s' % (name, resp.status_code))
```

Comment:
I had the print statements in for debugging purposes. I don't care weather we keep them or not :)